code/game/objects/machinery/minelayer.dm

```diff
@@ -0,0 +1,87 @@
+/obj/item/minelayer
+	name = "\improper M21 APRDS \"Minelayer\""
+	desc = "Anti-Personnel Rapid Deploy System, APRDS for short, is a device designed to quickly deploy M20 mines in large quantities. WARNING: Operating in tight places or existing mine fields will result in reduced efficiency."
+	icon = 'icons/Marine/marine-items.dmi'
+	icon_state = "minelayer"
+
+	max_integrity = 200
+	flags_item = IS_DEPLOYABLE|DEPLOYED_NO_ROTATE
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/minelayer/Initialize()
+	. = ..()
+	AddElement(/datum/element/deployable_item, /obj/machinery/deployable/minelayer, 1 SECONDS)
+
+/obj/machinery/deployable/minelayer
+	anchored = TRUE
+	density = TRUE
+	layer = ABOVE_MOB_LAYER
+	use_power = NO_POWER_USE
+	///how many mines are currently stored
+	var/stored_amount = 0
+	///how many mines can be stored
+	var/max_amount = 10
+	///radius on mine placement
+	var/range = 4
+
+/obj/machinery/deployable/minelayer/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+
+	playsound(loc, 'sound/machines/click.ogg', 25, 1)
+
+	sleep(2 SECONDS)
+
+	var/list/turf_list = list()
+
+	for(var/turf/T AS in orange(range, loc))
+		turf_list += T
+
+	while(stored_amount > 0 && turf_list.len > 0)
+		var/turf/target_turf = pick_n_take(turf_list)
+		if(!target_turf.density && !turf_block_check(src, target_turf) && !(locate(/obj/item/explosive/mine) in range(1, target_turf)))
+			var/obj/item/explosive/mine/placed_mine = new /obj/item/explosive/mine(loc)
+			placed_mine.throw_at(target_turf, range * 2, 1, src, TRUE)
+			playsound(loc, 'sound/machines/switch.ogg', 25, 1)
```

Comment:
This sound isnt very fitting consider maybe using a more fitting sound, here are some examples

https://freesound.org/people/LeMudCrab/sounds/163458/
https://freesound.org/people/qubodup/sounds/162401/
https://freesound.org/people/qubodup/sounds/182792/
https://freesound.org/people/jackyyang09/sounds/432781/